tracer/src/Datadog.Trace/Iast/VulnerabilityBatch.cs

```diff
@@ -87,27 +82,30 @@ private static JsonSerializerSettings CreateTruncatedSettings()
 
     public void Add(Vulnerability vulnerability)
     {
-        _vulnerabilitiesJson = null;
-        var ranges = vulnerability.Evidence?.Ranges;
-
-        if (ranges != null)
+        lock (Vulnerabilities)
```

Comment:
It looks like there's lots more potential concurrency issues in this type🤔 For example, there's concurrent access to `Sources` which similarly could cause issues

Reply:
Agree, but they only occur when adding a vulnerability when serializing, so covering both methods should be enough

Reply:
Fixed